lib/index.js

```diff
@@ -17,6 +17,24 @@ const validatorFactory = (handler, schema) => (payload, metadata) => { // eslint
     return handler(value, metadata);
 };
 
+const createExchange = (config, workerConfig, exchangeMap) => {
+
+    if (workerConfig.exchange) {
+        return workerConfig.exchange;
+    }
+
+    const exchangeName = workerConfig.exchangeName || config.defaults.exchangeName || workerConfig.name; // using the same fallbacks as minion lib
+    const exchangeType = workerConfig.exchangeType;
+    const exchangeMapKey = `${exchangeType}.${exchangeName}`;
+
+    if (exchangeMap[exchangeMapKey]) {
```

Comment:
Totally works fine this way, but what are your thoughts on:

```suggestion
    if (!exchangeMap[exchangeMapKey]) {
        exchangeMap[exchangeMapKey] = config.connection.rabbit[exchangeType](exchangeName);
    }
```
This way, you can just do `return exchangeMap[exchangeMapKey];` once.